src/api/app.py

```diff
@@ -980,7 +992,40 @@ async def chat(
     else:
         # For anonymous users, use the thread_id from request but don't persist
         # This allows conversation continuity within the same session
-        thread_id = request.thread_id
+        thread_id = chat_request.thread_id
+
+    # Populate langfuse metadata
+    langfuse_metadata = {}
+
+    identity = None
+
+    if user:
+        identity = user.email
+    else:
+        identity = await extract_anonymous_session_cookie(request)
+
+    # Note: should this be user.email or user.id?
+    langfuse_metadata["langfuse_user_id"] = identity
+
+    langfuse_metadata["langfuse_session_id"] = thread_id
+
+    # store additional user metadata if user is logged in
+    if user:
+        langfuse_metadata["user_internal_id"] = user.id
+
+        for attr in [
+            "profile_description",
+            "sector_code",
+            "role_code",
+            "job_title",
+            "company_organization",
+            "country_code",
+            "preferred_language_code",
+            "gis_expertise_level",
+            "areas_of_interest",
+            "has_profile",
+        ]:
+            langfuse_metadata[attr] = getattr(user, attr, None)
```

Comment:
i dont think we need to add all this user information for all the langfuse traces? I believe just user id should be enough?

Reply:
For sure! I was just being a data hoarder - but after today's discussion on privacy and PII I've removed it!